platform/src/main/java/org/stellar/anchor/platform/event/KafkaSession.java

```diff
@@ -0,0 +1,162 @@
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.Producer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.header.internals.RecordHeader;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.apache.kafka.common.serialization.StringSerializer;
+import org.springframework.kafka.support.serializer.JsonDeserializer;
+import org.springframework.kafka.support.serializer.JsonSerializer;
+import org.stellar.anchor.api.event.AnchorEvent;
+import org.stellar.anchor.api.exception.AnchorException;
+import org.stellar.anchor.api.exception.EventPublishException;
+import org.stellar.anchor.event.EventService;
+import org.stellar.anchor.event.EventService.EventQueue;
+import org.stellar.anchor.platform.config.KafkaConfig;
+import org.stellar.anchor.util.Log;
+
+public class KafkaSession implements EventService.Session {
+
+  final KafkaConfig kafkaConfig;
+  final String sessionName;
+  final String topic;
+  Producer<String, AnchorEvent> producer = null;
+  Consumer<String, AnchorEvent> consumer = null;
+
+  KafkaSession(KafkaConfig kafkaConfig, String sessionName, EventQueue queue) {
+    this.kafkaConfig = kafkaConfig;
+    this.sessionName = sessionName;
+    this.topic = queue.name();
+  }
+
+  @Override
+  public void publish(AnchorEvent event) throws AnchorException {
+    try {
+      if (producer == null) {
+        producer = createProducer();
+      }
+      ProducerRecord<String, AnchorEvent> record = new ProducerRecord<>(topic, event);
+      record.headers().add(new RecordHeader("type", event.getType().type.getBytes()));
+      // If the queue is offline, throw an exception
+      try {
+        producer.send(record).get();
+      } catch (Exception ex) {
+        throw new EventPublishException("Failed to publish event to Kafka.", ex);
+      }
+
+      // publish the event to the metrics
+      Metrics.counter(
+              "event.published",
+              "class",
+              event.getClass().getSimpleName(),
+              "type",
+              event.getType().type)
+          .increment();
+
+    } catch (Exception ex) {
+      Log.errorEx(ex);
+    }
+  }
+
+  @Override
+  public EventService.ReadResponse read() throws AnchorException {
+    if (consumer == null) {
+      consumer = createConsumer();
+      consumer.subscribe(java.util.Collections.singletonList(topic));
+    }
+
+    ConsumerRecords<String, AnchorEvent> consumerRecords =
+        consumer.poll(Duration.ofSeconds(kafkaConfig.getPollTimeoutSeconds()));
+    ArrayList<AnchorEvent> events = new ArrayList<>(consumerRecords.count());
+    if (consumerRecords.isEmpty()) {
+      Log.debugF("Received {} Kafka records", consumerRecords.count());
+    } else {
+      Log.infoF("Received {} Kafka records", consumerRecords.count());
+      for (ConsumerRecord<String, AnchorEvent> record : consumerRecords) {
+        events.add(record.value());
+      }
+    }
```

Comment:
Do we want to emit a metric for read as well?

Reply:
I put a TODO here. It will be handled in [ANCHOR-371](https://stellarorg.atlassian.net/browse/ANCHOR-371).